interfaces/builtin/udisks2.go

```diff
@@ -0,0 +1,918 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const udisks2PermanentSlotAppArmor = `
+# Description: Allow operating as the udisks2. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.UDisks2",
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UDisks2{,/**}
+    interface=org.freedesktop.DBus**
+    peer=(label=unconfined),
+
+# Needed for mount/unmount operations
+capability sys_admin,
+
+# Allow reading existing mount points
+@{PROC}/@{pid}/mountinfo r,
+@{PROC}/swaps r,
+
+# Udisks checks for existing configured mount locations here
+/etc/fstab r,
+
+# Allow scanning of devices
+network netlink raw,
+/run/udev/data/* r,
+/sys/devices/** r,
+
+# Mount points are in /run/media/<user>/* since /media is read-only
+/run/systemd/seats/* r,
+/run/media/ rw,
+/run/media/** rw,
+mount -> /run/media/**,
```

Comment:
`/media` vs `/run/media` is a build-time option in udisks, so the snap or a shared part could easily choose either. If `/media` is available now I'll change it though since that's probably what people expect.